pkg/registry/core/rest/storage_core.go

```diff
@@ -358,7 +358,7 @@ func (s componentStatusStorage) serversToValidate() map[string]*componentstatus.
 				klog.Errorf("Failed to split host/port: %s (%v)", etcdUrl.Host, err)
 				continue
 			}
-			port, _ = strconv.Atoi(portString)
+			port, _ = utilsnet.ParsePort(portString, false)
```

Comment:
is 0 allowed? previously it seems it was

Reply:
Yes, it does not care the range of the port previously, even for negative. And according to wikipedia `A port number is a 16-bit unsigned integer, thus ranging from 0 to 65535. For TCP, port number 0 is reserved and cannot be used, while for UDP, the source port is optional and a value of zero means no port.` I am not sure whether there is any special case that we need a zero port here. And across the codebase, there are methods like https://github.com/kubernetes/kubernetes/blob/b6f11382f5d63313b968028719e54a1afb3fe1ba/staging/src/k8s.io/apimachinery/pkg/util/validation/validation.go#L271-L277.

Reply:
you may want to use 0 if you want "any" available free port
https://stackoverflow.com/a/43425461/7794348

Reply:
Ok, I will make 0 available for these ports.

Reply:
sorry, I wasn't advocating for one or the other option, just that we should check if the code is expecting a 0 or not, otherwise, we'll modify current behavior

Reply:
`strconv.Atoi` allows a 0, so allowing 0 here at least will not make things worse, right?

Reply:
that was my thought too :smile: